engine/access/rest_api_test.go

```diff
@@ -0,0 +1,177 @@
+	suite.execClient = new(accessmock.ExecutionAPIClient)
+
+	suite.request = new(module.Requester)
+	suite.request.On("EntityByID", mock.Anything, mock.Anything)
+
+	suite.me = new(module.Local)
+
+	accessIdentity := unittest.IdentityFixture(unittest.WithRole(flow.RoleAccess))
+	suite.me.
+		On("NodeID").
+		Return(accessIdentity.NodeID)
+
+	suite.chainID = flow.Testnet
+	suite.metrics = metrics.NewNoopCollector()
+
+	const anyPort = ":0" // :0 to let the OS pick a free port
+	config := rpc.Config{
+		UnsecureGRPCListenAddr: anyPort,
+		SecureGRPCListenAddr:   anyPort,
+		HTTPListenAddr:         anyPort,
+		RESTListenAddr:         anyPort,
+		ServeRESTAPI:           true,
+	}
+
+	suite.rpcEng = rpc.New(suite.log, suite.state, config, suite.collClient, nil, suite.blocks, suite.headers, suite.collections, suite.transactions,
+		nil, nil, suite.chainID, suite.metrics, 0, 0, false, false, nil, nil)
+	unittest.AssertClosesBefore(suite.T(), suite.rpcEng.Ready(), 2*time.Second)
+
+	// wait for the server to startup
+	assert.Eventually(suite.T(), func() bool {
+		return suite.rpcEng.RestApiAddress() != nil
+	}, 5*time.Second, 10*time.Millisecond)
+}
+
+func TestRestAPI(t *testing.T) {
+	suite.Run(t, new(RestAPITestSuite))
+}
+
+func (suite *RestAPITestSuite) TestRestAPICall() {
+
+	suite.Run("GetBlockByID - happy path", func() {
+
+		collections := unittest.CollectionListFixture(1)
+		block := unittest.BlockWithGuaranteesFixture(
+			unittest.CollectionGuaranteesWithCollectionIDFixture(collections),
+		)
+		suite.blocks.On("ByID", block.ID()).Return(block, nil).Once()
+
+		client := suite.restAPIClient()
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+		defer cancel()
+
+		blocks, resp, err := client.BlocksApi.BlocksIdGet(ctx, []string{block.ID().String()}, nil)
+		require.NoError(suite.T(), err)
+		assert.Equal(suite.T(), http.StatusOK, resp.StatusCode)
+		assert.Len(suite.T(), blocks, 1)
+		assert.Equal(suite.T(), block.ID().String(), blocks[0].Header.Id)
+	})
+
+	suite.Run("GetBlockByID with a non-existing block ID", func() {
+
+		nonExistingBlockID := unittest.IdentifierFixture()
+		suite.blocks.On("ByID", nonExistingBlockID).Return(nil, storage.ErrNotFound).Once()
+
+		client := suite.restAPIClient()
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+		defer cancel()
+
+		_, resp, err := client.BlocksApi.BlocksIdGet(ctx, []string{nonExistingBlockID.String()}, nil)
+		assertError(suite.T(), resp, err, http.StatusNotFound, "not found")
+	})
+
+	suite.Run("GetBlockByID with an invalid block ID", func() {
+
+		client := suite.restAPIClient()
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+		defer cancel()
+
+		const invalidBlockID = "invalid_block_id"
+		_, resp, err := client.BlocksApi.BlocksIdGet(ctx, []string{invalidBlockID}, nil)
+		assertError(suite.T(), resp, err, http.StatusBadRequest, fmt.Sprintf("invalid ID %s", invalidBlockID))
+	})
+}
```

Comment:
A case where the length of block IDs exceeds the maximum may be good for preventing regression.

Reply:
yup added that